src/rf/apps/workers/emr.py

```diff
@@ -0,0 +1,194 @@
+
+                # TODO: Move to ansible
+                # For debugging only.
+                # 'Market': 'SPOT',
+                # 'BidPrice': '0.15',
+
+                'Market': 'ON_DEMAND',
+                'InstanceCount': int(settings.AWS_EMR_INSTANCES),
+            },
+        ],
+    }
+
+    config_env_vars = {
+        'Classification': 'export',
+        'Properties': {
+            'GDAL_DATA': '/usr/local/share/gdal',
+            'LD_LIBRARY_PATH': '/usr/local/lib',
+            'PYSPARK_PYTHON': 'python27',
+            'PYSPARK_DRIVER_PYTHON': 'python27',
+        },
+    }
+
+    configurations = [
+        {
+            'Classification': 'hadoop-env',
+            'Configurations': [config_env_vars],
+        },
+        {
+            'Classification': 'spark-env',
+            'Configurations': [config_env_vars],
+        },
+        {
+            'Classification': 'yarn-env',
+            'Configurations': [config_env_vars],
+        }
+    ]
+
+    response = client.run_job_flow(
+        Name=settings.AWS_EMR_CLUSTER_NAME,
+        LogUri=logs_uri,
+        ReleaseLabel='emr-4.0.0',
+
+        # These roles are created when you manually launch an EMR cluster
+        # or by using the following command:
+        # > aws emr create-default-roles
+        ServiceRole='EMR_DefaultRole',
+        JobFlowRole='EMR_EC2_DefaultRole',
+
+        Applications=[
+            {
+                'Name': 'Spark',
+            }
+        ],
+        BootstrapActions=[
+            {
+                'Name': 'Install dependencies',
+                'ScriptBootstrapAction': {
+                    'Path': bootstrap_uri,
+                }
+            }
+        ],
+        Instances=instances,
+        Configurations=configurations,
+        Steps=get_steps(layer, status_queue),
+    )
+    log.info(response)
+
+
+def add_steps(layer, status_queue, cluster_id):
+    """
+    Add steps to an existing cluster.
+    cluster_id - Existing EMR cluster ID
+    """
+    client = boto3.client('emr')
+    response = client.add_job_flow_steps(
+        JobFlowId=cluster_id,
+        Steps=get_steps(layer, status_queue),
+    )
+    log.info(response)
+
+
+def get_steps(layer, status_queue):
+    spark_submit = [
+        'spark-submit',
+        '--deploy-mode',
+        'cluster',
+        '--driver-memory',
+        '2g',
+    ]
+
+    workspace = 's3://{}'.format(settings.AWS_WORKSPACE_BUCKET)
+    chunk_result = 's3://{}/{}.json'.format(settings.AWS_WORKSPACE_BUCKET,
+                                            layer.id)
+    chunk_uri = 's3://{}/chunk.py'.format(settings.AWS_ARTIFACTS_BUCKET)
+    mosaic_uri = 's3://{}/mosaic.jar'.format(settings.AWS_ARTIFACTS_BUCKET)
+
+    images = LayerImage.objects.filter(layer_id=layer.id)
+    images = [image.get_s3_uri() for image in images]
+
+    return [
+        {
+            'Name': 'Chunk',
+            # For debugging only.
+            # 'ActionOnFailure': 'CONTINUE',
```

Comment:
Good idea. I've created a new ansible variable to control this and set it to false by default for all environments.